src/util/export/tei/teiExporter.ts

```diff
@@ -77,32 +78,45 @@ const createTaxonomy = (annotations: SupabaseAnnotation[], document: any) => {
     [...existingTaxonomies].map((el) => el.getAttribute('xml:id'))
   );
 
-  // TODO verify if the generated ID already exists in the header
   const taxonomyId = generateRandomId(existingTaxonomyIds);
 
-  const distinctTags = new Set(
+  const distinctTags: (string | VocabularyTerm)[] = [...new Set(
     annotations.reduce<string[]>((all, annotation) => {
       const tagBodies = (annotation.bodies || []).filter(
         (b) => b.purpose === 'tagging' && b.value
       );
       return [...all, ...tagBodies.map((b) => b.value!)];
     }, [])
-  );
+  )].map(str => {
+    // Tags might be plaintext or semantic
+    try {
+      return JSON.parse(str) as VocabularyTerm;
+    } catch {
+      return str;
+    }
+  });
 
   // Note: we'll later want to use URIs for tag IDs. Slugifying
   // the label is going to be a fallback option
-  const idFromLabel = (label: string) =>
-    slugify(label, {
+  const idFromTerm = (term: string | VocabularyTerm, taxonomyId: string) => {
```

Comment:
For the TEI taxonomy element, we'll use the ID of the tag if any, or a slugified version of the label.